packages/elements/src/react/common/form/index.tsx

```diff
@@ -443,7 +459,17 @@ const FieldInner = React.forwardRef<FormFieldElement, FormFieldProps>((props, fo
       {...rest}
       ref={forwardedRef}
     >
-      {typeof children === 'function' ? children(fieldState) : children}
+      <RadixValidityState>
+        {validity => {
+          const enrichedFieldState = enrichFieldState(validity, fieldState);
+
+          return (
+            <ValidityStateContext.Provider value={validity}>
+              {typeof children === 'function' ? children(enrichedFieldState) : children}
+            </ValidityStateContext.Provider>
+          );
+        }}
+      </RadixValidityState>
```

Comment:
Main gist of this PR. Using Context here to not have to pass around values through props 👍